docs/concepts/SafeRollout/README.md

```diff
@@ -0,0 +1,214 @@
+    - group: ""
+      kind: Namespace
+      version: v1          
+      name: test-ns
+  policy:
+    placementType: PickN
+    numberOfClusters: 2
+    affinity:
+      clusterAffinity:
+        requiredDuringSchedulingIgnoredDuringExecution:
+          clusterSelectorTerms:
+              - labelSelector:
+                  matchLabels:
+                    loc: east
+  strategy:
+    rollingUpdate:
+      maxSurge: 2
+```
+
+The rollout will be as follows:
+
+- We try to pick clusters (cluster-3 and cluster-4) by specifying the label selector `loc: east`.
+- But this time around since we have `maxSurge` set to 2 we are saying we can propagate resources to a maximum of 
+4 clusters but our target number of clusters specified is 2, we will rollout the namespace with deployment to both 
+cluster-3 and cluster-4 before removing the deployment from cluster-1 and cluster-2. 
+- And since `maxUnavailable` is always set to 25% by default which is rounded off to 1, we will remove the 
+resource from one of the existing clusters (cluster-1 or cluster-2) because when `maxUnavailable` is 1 the policy 
+mandates at least one cluster to be available.
+
+### UnavailablePeriodSeconds
+
+`UnavailablePeriodSeconds` is used to configure the waiting time between rollout phases when we cannot determine if the 
+resources have rolled out successfully or not. This field is used only if the availability of resources we propagate 
+are not trackable. Refer to the [Data only object](#data-only-objects) section for more details.
+
+## Availability based Rollout
+We have built-in mechanisms to determine the availability of some common Kubernetes native resources. We only mark them 
+as available in the target clusters when they meet the criteria we defined.
+
+### How It Works
+We have an agent running in the target cluster to check the status of the resources. We have specific criteria for each 
+of the following resources to determine if they are available or not. Here are the list of resources we support:
+
+#### Deployment
+We only mark a `Deployment` as available when all its pods are running, ready and updated according to the latest spec. 
+
+#### DaemonSet 
+We only mark a `DaemonSet` as available when all its pods are available and updated according to the latest spec on all 
+desired scheduled nodes.
+
+#### StatefulSet
+We only mark a `StatefulSet` as available when all its pods are running, ready and updated according to the latest revision.
+
+#### Job
+We only mark a `Job` as available when it has at least one succeeded pod or one ready pod.
+
+#### Service
+For `Service` based on the service type the availability is determined as follows:
+
+- For `ClusterIP` & `NodePort` service, we mark it as available when a cluster IP is assigned.
+- For `LoadBalancer` service, we mark it as available when a `LoadBalancerIngress` has been assigned along with an IP or Hostname.
+- For `ExternalName` service, we don't know how to determine the availability.

```

Comment:
```suggestion
- For `ExternalName` service, checking availability is not supported, so it will be marked as available with not trackable reason.

```